apps/site/assets/ts/schedule/ScheduleLoader.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { ReactElement } from "react";
+import { connect } from "react-redux";
+import { useQueryParams, StringParam } from "use-query-params";
+import SchedulePage from "./components/SchedulePage";
+import ScheduleNote from "./components/ScheduleNote";
+import ScheduleDirection from "./components/ScheduleDirection";
+import { SchedulePageData, SelectedOrigin } from "./components/__schedule";
+import { MapData, StaticMapData } from "../leaflet/components/__mapdata";
+import ScheduleFinder from "./components/ScheduleFinder";
+import ScheduleFinderModal from "./components/schedule-finder/ScheduleFinderModal";
+import { DirectionId } from "../__v3api";
+import mapStateToProps, {
+  getCurrentState,
+  storeHandler
+} from "./ScheduleStore";
+
+interface Props {
+  schedulePageData: SchedulePageData;
+  component: string;
```

Comment:
@alicia-mbta this would be a good place to use a [string literal type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#string-literal-types). In this case it would be something like:

```
type ComponentType = "MAIN" | "SCHEDULE_NOTE" | "SCHEDULE_FINDER" | "SCHEDULE_DIRECTION"
```

and then you can define the `component` field here as having type `ComponentType`.